### ISSUE TITLE
Unable to create WindowedContext while EventLoop is running

### ISSUE THREAD

**Issue by fschutt** [rust-windowing/glutin#1184]
[The docs say](https://docs.rs/glutin/0.22.0-alpha1/glutin/event_loop/struct.EventLoopWindowTarget.html) that if you want to create a window while the application is running, you need to create a proxy and that `EventLoopWindowTarget` exists because "this type exists to allow you to create new windows while Winit executes your callback. EventLoop will coerce into this type, so functions that take this as a parameter can also take &EventLoop". The problem with that is that `ContextBuilder::build_windowed` and `ContextBuilder::build_headless` only accept a `&EventLoop`, not a `EventLoopWindowTarget` - and you need those functions to create a window + a context at runtime.

While, yes, you can create a `Window` at runtime (via `WindowBuilder::build()`), that only gives you the window. What I actually want is a `WindowedContext` (so that I can call .get_current() and draw on the window). Just having a window alone is fairly pointless without the context. So I think it's maybe an API oversight that `.build_windowed()` and `.build_headless()` take a `&EventLoop` while they should be taking a `EventLoopWindowTarget` instead - otherwise, there's no way to create new contexts to draw on while the app is running.

Ideally, this should work:

```rust
let el = glutin::event_loop::EventLoop::new();
let el_proxy = el.create_proxy(); // <--
let wb = glutin::window::WindowBuilder::new()
    .with_title("Hello world!")
    .with_inner_size(glutin::dpi::LogicalSize::new(1024.0, 768.0));
let windowed_context = glutin::ContextBuilder::new()
    .build_windowed(wb, el_proxy) // <--
    .unwrap();
```

**Comment by ZeGentzy** [comment]
This is indeed an oversight. Thank you.

**Comment by fschutt** [comment]
Isn't this fixed in https://github.com/rust-windowing/glutin/commit/e90641f1c5be3571e48d23051944b4f7dd6921f8 ?

**Comment by fschutt** [comment]
Yep, this has been fixed in `glutin-0.22-alpha3`. Thanks!
